designs/administrator/future-openshift/bulk-actions/index.md

```diff
@@ -0,0 +1,139 @@
+
+If the user has permission to perform a specific action on **some** of the selected items, the action will be carried out for all of the items that it can be. A failure alert will be included within the modal to list the items that failed to complete.
+![some-permissions](img/some-permissions.png)
+
+
+### Infinite scrolling
+If a list view does not contain pagination and includes infinite scrolling, when the user uses the bulk selector to select all items, only the items within view would be selected. 
+![infinite-scrolling-all-selected](img/infinite-scrolling-all-selected.png)
+
+Once the user scrolls down to load more items, they would have to use the bulk selector to select none then select all again to select all items that have loaded.
+![infinite-scrolling-page-down](img/infinite-scrolling-page-down.png)
+
+### Filtering
+If a list view contains filtering, items that have already been selected will remain selected if filters are applied or cleared. 
+![filter-clear](img/filter-clear.png)
+![filter-update](img/filter-update.png)
+
+They may select additional items from the filtered list and once they do, the item count in the bulk selector will update to reflect the action. 
+![filter-select](img/filter-select.png)
+
+## Actions
+Only specific actions will be available to take on multiple resources. The default bulk actions are `Add Labels`, `Add Annotations` and `Delete Resources`. Additional actions may be added in future depending on the resource type. When taking actions on five resources or fewer, the modal will list out the resource names. When taking actions on more than five resources, the modal will simply list the number (e.g. “Labels for 25 resources.”).
+
+### Add Labels
+The `Add Labels` modal allows the user to add labels to multiple resources and notes that existing labels may be replaced if there are duplicates. Users can add labels in the textbox and click ‘Save’ to add the labels to the selected resources. The new labels will appear in the list and the items will remain selected after the modal closes.
+![add-labels-modal](img/add-labels-modal.png)
+
+### Add Annotations
+The `Add Annotations` modal allows the user to add annotations to multiple resources and notes that existing annotations may be replaced. Users can add annotations and click ‘Save’ to add the annotations to the selected resources. The items will remain selected after the modal closes.
+![add-annotations-modal](img/add-annotations-modal.png)
+
+### Delete Resources
+
+#### No failures
+
+The `Delete Resources` modal will list out the resources that are being chosen to be deleted and will be checked by default. There will be a bulk selector above the listed items that will showcase the item count and will act the same way that the bulk selector within the toolbar works. The user may change these checkboxes if desired, but this acts as a “second check” for them. The user must select “Delete” in order to begin the deletion process.
```

Comment:
I sort of feel like the bulk selection in the bulk selection is maybe overly inception-like to me and the dialog could just list out the selections? Like if they really mis-selected something it doesn't seem to onerous to ask them to exit the modal and deselect it from the list? Just a thought.

Reply:
Since they're deleting resources, we wanted them to have an extra check here - originally there was the idea of typing "delete" in a text box to confirm. I think I went this direction after one of your designs with deleting operator resources or something? Does that ring a bell? @itsptk

Reply:
Ah yeah, that was when an operator was being deleted, also giving the option to delete the resources (operands) that had been created with it while it was installed: https://docs.google.com/document/d/1Pwb0ffOq45BdtsdZvRt5inIJi96ASDp0c4Rqy4vW3QM/edit?usp=sharing

I guess I'd say that case was sort of "you are deleting this thing, would you also want to delete these related things" whereas this bulk example is the same list of selected items. I think I am definitely for listing out the resources they selected and will be deleted, I just wasn't sure if they still needed to be actionable at that point.

Reply:
It's more of a chance for the user to easily be able to get rid of one of the listed resources if they needed to without have to cancel the whole thing and go back to the list view! I'll make the text clearer but they wouldn't be able to take any action to these items from this - only select or deselect to confirm those are the ones they want to delete.